library/src/main/java/com/michaldrabik/classicmaterialtimepicker/CmtpDateDialogFragment.kt

```diff
@@ -0,0 +1,123 @@
+      positiveButtonText: String = "OK",
+      negativeButtonText: String = "Cancel"
+    ) = CmtpDateDialogFragment().apply {
+      arguments = Bundle().apply {
+        putString(ARG_POSITIVE_BUTTON_TEXT, positiveButtonText)
+        putString(ARG_NEGATIVE_BUTTON_TEXT, negativeButtonText)
+      }
+    }
+  }
+
+  private lateinit var date: CmtpDate
+  private lateinit var customYearRange: IntRange
+  private lateinit var customDateSeparator: String
+  private lateinit var datePicker: CmtpDatePickerView
+
+  private lateinit var onDatePickedListener: OnDatePickedListener
+
+  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+    val context = requireContext()
+
+    datePicker = CmtpDatePickerView(context)
+    savedInstanceState?.let { restoreState(it) }
+    if (this::date.isInitialized) datePicker.setDate(date)
+    if (this::customYearRange.isInitialized) datePicker.setCustomYearRange(customYearRange)
+    if (this::customDateSeparator.isInitialized) datePicker.setCustomDateSeparatorCharacter(customDateSeparator)
+
+    val dialogBuilder = AlertDialog.Builder(context, R.style.CmtpDialogFrameStyle)
+    dialogBuilder.setView(datePicker)
+
+    dialogBuilder.setPositiveButton(
+      arguments?.getString(ARG_POSITIVE_BUTTON_TEXT)
+    ) { _, _ -> onDatePicked() }
+
+    dialogBuilder.setNegativeButton(
+      arguments?.getString(ARG_NEGATIVE_BUTTON_TEXT)
+    ) { _, _ -> }
+
+    return dialogBuilder.create()
+  }
+
+  private fun saveState(outState: Bundle) {
+    if (!this::datePicker.isInitialized) return
+    outState.putInt(ARG_DAY, datePicker.getDate().day)
+    outState.putInt(ARG_MONTH, datePicker.getDate().month)
+    outState.putInt(ARG_YEAR, datePicker.getDate().year)
+  }
+
+  private fun restoreState(stateBundle: Bundle) {
+    val day = stateBundle.getInt(ARG_DAY, CmtpDate.DEFAULT.day)
+    val month = stateBundle.getInt(ARG_MONTH, CmtpDate.DEFAULT.month)
+    val year = stateBundle.getInt(ARG_YEAR, CmtpDate.DEFAULT.year)
+
+    date = CmtpDate(day, month, year)
+  }
+
+  override fun onResume() {
+    super.onResume()
+    dialog.window?.let {
+      val width = resources.getDimensionPixelSize(R.dimen.cmtp_timepicker_width)
+      it.setLayout(width, WRAP_CONTENT)
+    }
+  }
+
+  private fun onDatePicked() {
+    if (this::onDatePickedListener.isInitialized) {
+      onDatePickedListener.onDatePicked(datePicker.getDate())
+    }
+  }
+
+  /**
+   * Set initial date and initialize picker
+   * @param day from 1 to 31 (depending on month).
+   * @param month from 1 to 12.
+   * @param year from currentYear-100 to currentYear+100
+   * @throws IllegalStateException when given day, month or year is out of valid range.
+   */
+  fun setInitialDate(day: Int, month: Int, year: Int) {
+    date = CmtpDate(day, month, year)
+  }
+
+  fun setCustomYearRange(startingYear: Int, endingYear: Int) {
+    customYearRange = (startingYear..endingYear)
+  }
+
+  fun setCustomDateSeparatorCharacter(separator: String) {
```

Comment:
`setCustomSeparator()` name will be enough.

Reply:
Ok